python/paddle/fluid/dygraph/dygraph_to_static/list_transformer.py

```diff
@@ -283,20 +218,44 @@ def _update_list_name_to_updated(self, node):
             del self.list_name_to_updated[target_id]
         return False
 
-    def _replace_list_pop(self, node):
+    def _replace_pop(self, node):
+        """
+        Replace a pop statement for a list or dict.
+        For example:
+
+            list_a = [0,1,2,3,4]
+            x = list_a.pop()  # --> convert_pop(list_a)
+            y = list_a.pop(1) # --> convert_pop(list_a, 1)
+
+            dict_a = {"red":0, "blue":1, "yellow":2}
+            m = dict_a.pop("red")           # --> convert_pop(dict_a, "red")
+            n = dict_a.pop("black", 3)      # --> convert_pop(dict_a, "black", 3)
+
+        """
         assert isinstance(node, gast.Call)
         assert isinstance(node.func, gast.Attribute)
 
         target_node = node.func.value
         target_str = ast_to_source_code(target_node).strip()
 
-        if node.args:
-            idx_node = node.args[0]
-            idx_str = ast_to_source_code(idx_node).strip()
+        args_str = [ast_to_source_code(arg).strip() for arg in node.args]
+
+        # 1. pop stmt for a list
+        if len(args_str) == 0:
+            new_call_str = "paddle.jit.dy2static.convert_pop({})".format(
+                target_str, args_str)
+
+        # 2. pop stmt for a list or dict
+        elif len(args_str) == 1:
+            new_call_str = "paddle.jit.dy2static.convert_pop({}, {})".format(
+                target_str, args_str[0])
+
+        # 3. pop stmt for a dict
+        elif len(args_str) == 2:
+            new_call_str = "paddle.jit.dy2static.convert_pop({}, {}, {})".format(
+                target_str, args_str[0], args_str[1])
```

Comment:
why not use `new_call_str = "paddle.jit.dy2static.convert_pop({})".format(target_str, *args_str)` directly? Because your `convert_pop` is defined as `(target, *args)`

Reply:
Thanks, done